piksi_tools/ardupilot/mavlink_decode.py

```diff
@@ -8,92 +8,273 @@
+      start_index = end_index
+      end_index = start_index + FORMAT_SIZE_BYTES[self.FORMAT_HEADER[1]]
+      self.msg_type = unpack('<' + self.FORMAT_HEADER[1], binary[start_index:end_index])[0]
+      start_index = end_index
+      end_index = start_index + FORMAT_SIZE_BYTES[self.FORMAT_HEADER[2]]
+      self.sender_id = unpack('<' + self.FORMAT_HEADER[2], binary[start_index:end_index])[0]
+      start_index = end_index
+      end_index = start_index + FORMAT_SIZE_BYTES[self.FORMAT_HEADER[3]]
+      self.msg_len = unpack('<' + self.FORMAT_HEADER[3], binary[start_index:end_index])[0]
+      self.msg = binary[end_index:]
+
+
+class SBR2:
+
+  HEADER = bytearray([0xA3, 0x95, 0xE6])
+  LABELS = "TimeUS,msg_type,1,2,3,4,5,6,7,8,9,10,11,12,13"
+  FORMAT_HEADER = 'QH'  # Data format before binary message
+  FORMAT_BINARY = 'QQQQQQQQQQQQQ'  # Data format for binary message
+  SIZE = 114
+  BINARY_SIZE = 104
+
+  def __init__(self, binary):
+    self.time_us = None
+    self.msg_type = None
+    self.msg = None
+    self.read_bytes(binary)
+
+  def read_bytes(self, binary):
+    if len(binary) == self.SIZE:
+      start_index = 0
+      end_index = FORMAT_SIZE_BYTES[self.FORMAT_HEADER[0]]
+      self.time_us = unpack('<' + self.FORMAT_HEADER[0], binary[start_index:end_index])[0]
+      start_index = end_index
+      end_index = start_index + FORMAT_SIZE_BYTES[self.FORMAT_HEADER[1]]
+      self.msg_type = unpack('<' + self.FORMAT_HEADER[1], binary[start_index:end_index])[0]
+      self.msg = binary[end_index:]
+
+
+class GPS:
+  HEADER = bytearray([0xA3, 0x95, 0x82])
+  HEADER2 = bytearray([0xA3, 0x95, 0x83])
+  FORMAT = "QBIHBcLLefffB"
+  LABELS = "TimeUS,Status,GMS,GWk,NSats,HDop,Lat,Lng,Alt,Spd,GCrs,VZ,U"
+  SIZE = 43
+
+  def __init__(self, binary):
+    self.time_us = None
+    self.gms = None
+    self.gwk = None
+    self.read_bytes(binary)
+
+  def read_bytes(self, binary):
+    if len(binary) == self.SIZE:
+      start_index = 0
+      end_index = FORMAT_SIZE_BYTES[self.FORMAT[0]]
+      self.time_us = unpack('<' + self.FORMAT[0], binary[start_index:end_index])[0]
+      start_index = end_index
+      end_index = start_index + FORMAT_SIZE_BYTES[self.FORMAT[1]]
+      # DO NOTHING CAUSE WE DON'T CARE OF STATUS
+      start_index = end_index
+      end_index = start_index + FORMAT_SIZE_BYTES[self.FORMAT[2]]
+      self.gms = unpack('<' + self.FORMAT[2], binary[start_index:end_index])[0]
+      start_index = end_index
+      end_index = start_index + FORMAT_SIZE_BYTES[self.FORMAT[3]]
+      self.gwk = unpack('<' + self.FORMAT[3], binary[start_index:end_index])[0]
+
+
+def get_first_gps_message(filename):
+  with open(filename, "rb") as log:
+    while search_binary_key(log, [GPS.HEADER, GPS.HEADER2]):
+      binary = log.read(GPS.SIZE)
+      assert len(binary) == GPS.SIZE, "Length of GPS message inconsistent"
+      gps = GPS(binary)
+      if gps.gwk and gps.gms:
+        return gps
+
+
+def gps_time_to_utc(gps_week, gps_milliseconds, time_us):
```

Comment:
I don't think that matters because this just puts some kind of time into the json metadata decorator, but we shouldn't call it UTC because it isn't UTC per say.